C++/include/algorithm/dynamic_programming/weighted_activity_selection.hpp

```diff
@@ -0,0 +1,125 @@
+/*
+    Weighted activity selection problem
+    --------------------
+	Given N activities where every activity is represented by 
+	following three elements of it.
+
+	1. Start Time
+	2. Finish Time
+	3. Weight or Value Associated
+
+	Find the maximum weight subset of activities such that no two of them in the subset overlap.
+
+
+    Time complexity
+    ---------------
+    O(N*lg(N))
+
+    Space complexity
+    ----------------
+    O(N)
+
+    Author
+    ------
+    Fedurko Nikolaus (@B1Z0N)
+*/
+
+#include <ios>
+#include <vector>
+#include <algorithm>
+
+#ifndef WEIGHTED_ACTIVITY_SELECTION_HPP
+#define WEIGHTED_ACTIVITY_SELECTION_HPP
+
+/**
+ * Struct for single activity data
+ */
+struct Activity
+{
+	int start;
+	int end;
+	int weight;
+};
+
+/**
+ * Find the index of right-most 
+ * non-overlaping activity
+ * that is left to the query[ index ]
+ * using binary search in O( lg(index) )
+ */
+int left_activity_bsearch( const std::vector<Activity> quer, int index )
```

Comment:
Pretty sure `quer` can be a reference. Also, might want to pick a clearer variable name.